float64 r1
float64 r2
float64 r3
float64 l
float64 vol
float64 weight
